src/TrainDialog.ts

```diff
@@ -45,6 +46,8 @@ export interface TrainScorerStep {
   importText?: string
   // Used for UI rendering only
   uiScoreResponse?: ScoreResponse
+  // If set, this action should be created with the specified condition(s).
```

Comment:
Does this field exist only for .dialog import?  If so, it doesn't feel right to add it to the core object if it's never used again.  Might be better to create a temporary object (i.e. perhaps a Map between the scorer step and required conditions)

Reply:
Yes, currently this is only for .dialog import.  This seems like something that would be more generally useful, though, since action generation happens in a separate step from any kind of import.  Thoughts ?

Reply:
Yea, I would agree is confusing to add it to this object, but it's been done for a bunch of others `importHashes` and `enumValues` etc so idk.

I don't have much context on reasoning for as you seemed to have acknowledged `requiredConditions` on actions and in the description but yet still decided this field was needed.  It seems these TrainScorerSteps would have to be scrubbed later so this train score step with required conditions is never in the same object as actions which also had required conditions and they wonder which take precedent and why there are duplicates.

I see do alternatives:
1. Create `ImportScorerStep` only for import and has fields for import. After import is doens convert everything to normal.
2. Save `requiredConditions` on a temporary Action object linked by `labelAction` ID to associate them with scorerStep then do whatever

Reply:
One difference with importHashes it that it continues to be used for the lifetime of the model as can add additional transcripts in the future.  .dialog import is one time only

Reply:
I'm not wedded to the placement here, was just trying to keep the conditions close to the thing that uses them and will be passed to the action creator.  Would you guys prefer a map/object to be set in the TrainDialog state ?

Reply:
At a minimum I suggest the variable name should be changed to indicate that it's only used for import.  Also, I think the server will throw an error as this is a unknown field.

Reply:
This works on the server; see screenshots from the actual import in https://github.com/microsoft/ConversationLearner-UI/pull/1347

I'm fine moving this to a map in TrainDialog state, but I think we'd still need to add an `id` to the `TrainScorerStep` if we went with that approach.  I don't think we can uniquely identify a scorer step in the tree just by hashing its content.